geniusyield-market-maker/src/GeniusYield/MarketMaker/Prices/Taptools.hs

```diff
@@ -0,0 +1,91 @@
+module GeniusYield.MarketMaker.Prices.Taptools where
+
+import           Data.Aeson
+import           Data.Aeson.Types                 (typeMismatch)
+import qualified Data.Map                         as Map
+import           Data.Maybe                       (fromJust)
+import           Data.Proxy
+import qualified Data.Text                        as Text
+import           Data.Text.Encoding               (encodeUtf8)
+import           Data.Time.Clock.POSIX            (POSIXTime)
+import           GeniusYield.GYConfig             (Confidential (..))
+import           GeniusYield.Imports              (Generic)
+import           GeniusYield.MarketMaker.Utils
+import           Network.HTTP.Client              (newManager, Manager, ManagerSettings(..), Request(..))
+import           Network.HTTP.Client.TLS          (tlsManagerSettings)
+import           Servant.API
+import           Servant.Client
+
+
+-- | Taptools time resolution for OHLC Candles
+data TtResolution = TtRes3m | TtRes5m | TtRes15m | TtRes30m | TtRes1h | TtRes2h | TtRes4h | TtRes12h
+                  | TtRes1d | TtRes3d | TtRes1w  | TtRes1mo
+                  deriving stock (Eq, Ord, Generic)
+
+instance FromJSON TtResolution where
```

Comment:
Use automatic `FromJSON` derivation, like done in other places: https://github.com/maestro-org/haskell-sdk/blob/609515b3f8861efaa17adf4e599aa1b845775a6d/src/Maestro/Types/V1/Common.hs#L89-L91.

Reply:
Done.

Reply:
Have a look at automatic derivation here: https://github.com/maestro-org/haskell-sdk/blob/609515b3f8861efaa17adf4e599aa1b845775a6d/src/Maestro/Types/V1/DefiMarkets.hs#L45-L48.

Reply:
Done.